diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index b9137e1..c80adbf 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -6,7 +6,8 @@ sub vc_win64a_info {
     unless (%$vc_win64a_info) {
         # Minimum NASM version is 2.09 otherwise SHA3 might be miscompiled
-        if (`nasm -v 2>NUL` =~ /NASM version ([0-9]+)\.([0-9]+)/ && ($1 > 2 || ($1 == 2 && $2 >= 9))) {
-            $vc_win64a_info = { AS        => "nasm",
+        my $voxflow_nasm = $ENV{VOXFLOW_OPENSSL_NASM} || "nasm";
+        if (`"$voxflow_nasm" -v 2>NUL` =~ /NASM version ([0-9]+)\.([0-9]+)/ && ($1 > 2 || ($1 == 2 && $2 >= 9))) {
+            $vc_win64a_info = { AS        => $voxflow_nasm,
                                 ASFLAGS   => "-g",
                                 asflags   => "-Ox -f win64 -DNEAR",
                                 asoutflag => "-o ",
diff --git a/Configurations/windows-makefile.tmpl b/Configurations/windows-makefile.tmpl
index 985bf46..73de650 100644
--- a/Configurations/windows-makefile.tmpl
+++ b/Configurations/windows-makefile.tmpl
@@ -794,9 +794,10 @@ EOF
           }
 
           if (defined($generator)) {
+              my $voxflow_cmd = $ENV{VOXFLOW_OPENSSL_CMD} || "cmd";
               return <<"EOF";
 $target: "$gen0" $deps
-	cmd /C "set "ASM=\$(AS)" & $generator \$@"
+	"$voxflow_cmd" /C "set "ASM=\$(AS)" & $generator \$@"
 EOF
           }
           return <<"EOF";
